apis/flinkcluster/v1beta1/flinkcluster_types.go

```diff
@@ -336,6 +345,11 @@ type TaskManagerSpec struct {
 	// [More info](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims)
 	VolumeClaimTemplates []corev1.PersistentVolumeClaim `json:"volumeClaimTemplates,omitempty"`
 
+	// _(Optional)_ Defines the mounted volume's storage type. If not specified, the default is `Persistent`.
+	// Persistent storage type is used with Statefulsets for stateful recovery.
+	// Ephemeral storage type is used with Deployments for faster startup.
+	StorageType StorageType `json:"storageType,omitempty"`
```

Comment:
The thing is that you can use that with StatefulSet's sooo ... would it be better actually to just use a deployment type property? I'm leaning towards this actually.

Reply:
Yes. Will be more clear to just use Deployment/Statefulset in the newly introduced property. 

In case of `kind: Deployment` however, I still want to take the volumeClaims from the TaskManagerSpec and append them to Ephemeral volumes in the Deployment. This way it will be easier for users to switch between the two `kinds`.

Wdyt?

Reply:
Yeah I guess we can do that; basically appending to the volumes; We need to be explicit about that in the docs though.

Reply:
Yeah I guess we can do that; basically appending to the volumes; We need to be explicit about that in the docs though.

Reply:
Will add more comments next to volumeClaim too 👍